plugins/auth-backend-module-microsoft-provider/src/strategy.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassportProfile } from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import fetch from 'node-fetch';
+import { Strategy as MicrosoftStrategy } from 'passport-microsoft';
+
+export class ExtendedMicrosoftStrategy extends MicrosoftStrategy {
+  userProfile(
+    accessToken: string,
+    done: (err?: Error | null, profile?: PassportProfile) => void,
+  ): void {
+    if (this.skipUserProfile(accessToken)) {
+      done(null, undefined);
+      return;
+    }
+
+    super.userProfile(
+      accessToken,
+      async (err?: Error | null, profile?: PassportProfile) => {
+        if (profile) {
+          profile.photos =
+            profile.photos ?? (await this.getProfilePhotos(accessToken));
```

Comment:
Can't `await` here, gonna need to chain with `.then` into `done` instead. If this throws it's gonna be an unhandled promise rejection and this request gets stuck

Reply:
That makes sense, I will adjust it.